editor/plugins/tiles/tile_map_editor.cpp

```diff
@@ -180,11 +187,19 @@ void TileMapEditorTilesPlugin::_update_tile_set_sources_list() {
 		}
 
 		sources_list->add_item(item_text, texture);
-		sources_list->set_item_metadata(i, source_id);
+		sources_list->set_item_metadata(sources_list->get_item_count() - 1, source_id);
 	}
 
 	if (sources_list->get_item_count() > 0) {
-		if (old_current > 0) {
+		if (prev_source >= 0) {
+			for (int i = 0; i < sources_list->get_item_count(); i++) {
+				if ((int)sources_list->get_item_metadata(i) == prev_source) {
+					sources_list->set_current(i);
+					sources_list->ensure_current_is_visible();
+					break;
+				}
+			}
+		} else if (old_current > 0) {
```

Comment:
I think this can be removed. We used to rely on the selection index to select the right source, but if we use the metadata instead, there's no need to keep the index-based logic.